hoodie-common/src/main/java/com/uber/hoodie/common/table/log/HoodieLogFormatWriter.java

```diff
@@ -127,10 +133,26 @@ public Writer appendBlock(HoodieLogBlock block)
     return rolloverIfNeeded();
   }
 
+  private boolean isLogFileSealed(Path path) throws IOException {
+    long currentOffset = this.output.getPos();
+    FSDataInputStream input = null;
+    try {
+      input = fs.open(path);
+      input.readFully(currentOffset - magicBuffer.length, magicBuffer);
+      if (Arrays.equals(magicBuffer, HoodieLogFormat.END_MARKER)) {
+        return true;
+      }
+      return false;
+    } finally {
+      input.close();
+    }
+  }
+
   private Writer rolloverIfNeeded() throws IOException, InterruptedException {
     // Roll over if the size is past the threshold
     if (getCurrentSize() > sizeThreshold) {
-      //TODO - make an end marker which seals the old log file (no more appends possible to that file).
+      //End marker which seals the old log file (no more appends possible to this file).
+      this.output.write(HoodieLogFormat.END_MARKER);
```

Comment:
What happens if the rolling over fails and the file max size is changed before the next append. 
I think this should be done after rolling over successfully to ensure we have atleast one open log for every file.

Reply:
Good point, made necessary changes.